### PR TITLE
Run Kiali tests in OSSMC

### PR DESCRIPTION
**Pull request by ferhoyos**
### Describe the change

When we reused Kiali's source code as the foundation for the OSSMC plugin, we assumed that Kiali's Cypress tests would also be valid for OSSMC. In that case, we only needed to create Cypress tests for specific OSSMC pages, such as the Istio config list page or specific navigation scenarios. However, recent issues (see[ issue #330](https://www.github.com/kiali/openshift-servicemesh-plugin/issues/330)) have shown that the same test scenarios can behave differently in Kiali and OSSMC.

To address this, we have modified the OSSMC Cypress test infrastructure to run the same Kiali test scenarios in OSSMC. We have introduced the `@ossmc` Cypress tag to run only those Kiali tests that are applicable to OSSMC (excluding multi-cluster, wizards, etc.).

Actions performed to enable running Kiali tests:
- Defined the same environment variables used in Kiali Cypress tests (`USERNAME`, `PASSWD`, `AUTH_PROVIDER`, etc.)
- Created the `@ossmc` Cypress tag to include a Cypress test in the OSSMC test suite.
- Created the `@skip-ossmc` Cypress tag to exclude specific Cypress tests from features included in OSSMC.
- Organized Kiali and OSSMC Cypress tests into their respective test folders (similar structure to the code, with`kiali` and `openshift` folders)
- Overrode the `visit` and `request` Cypress commands since there are slight differences in page URLs and APIs between Kiali and OSSMC.

### Steps to test the PR

1. Execute `yarn cypress:run` and verify that OSSMC cypress test pass

### Issue reference

https://github.com/kiali/openshift-servicemesh-plugin/issues/356

### CODE REVIEW

**Comment by ScriptingShrimp** [comment]
testing via upstream-istio-pipeline/2165

File: plugin/cypress/README.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ Before you start using Cypress suite, you might need export some environment var
 
 ```bash
 export CYPRESS_BASE_URL=<value>               # defaults to http://localhost:9000
-export CYPRESS_OC_CLUSTER_USER=<value>        # defaults to jenkins, opt. kubeadmin
-export CYPRESS_OC_CLUSTER_PASS=<value>        # no defaults
-export CYPRESS_OC_IDP=<value>                 # defaults to my_htpasswd_provider
+export CYPRESS_USERNAME=<value>               # defaults to jenkins, opt. kubeadmin
```
Comment (ScriptingShrimp) [inline plugin/cypress/README.md:15]: Originally I chose different variable names to be double sure that settings from kiali/kiali will not interfere with OSSMC suite

File: plugin/cypress.config.ts [hunk 2]
```diff
@@ -18,13 +18,14 @@ export default defineConfig({
   videosFolder: 'cypress/videos',
 
   env: {
-    OC_CLUSTER_USER: 'jenkins', // default value for jenkins
-    OC_IDP: 'my_htpasswd_provider', // default value for jenkins, can vary based on cluster setup
     'cypress-react-selector': {
       root: '#root'
     },
+    OSSMC: true,
```
Comment (ScriptingShrimp) [inline plugin/cypress.config.ts:24]: only nitpick, but this is why I would prefer to have two different configs, so there is no risk for them to be overwritten when copying the code

**Review by ScriptingShrimp — APPROVED** [review]
This is great!

File: plugin/cypress.config.ts (hunk 2 above)
Comment (ferhoyos) [reply plugin/cypress.config.ts:24]: Actually there are two configs, the `cypress.config` from Kiali repo and `cypress.config` from OSSMC repo. What I don't see is to have two configs just for OSSMC.

**Pull request merged**